### PR TITLE
Add swap() tests

### PR DESCRIPTION
**Pull request by uri-99**
This PR resolves issue #149, adding a suite of tests for the swap() functionality.

This PR also solves issue #162 , as a pool test case was necesary to build the structure for testing the different pools

This PR also solves a minor bug found on scripts/local.rs , where the BUILD_PATH_PREFIX was incomplete

### CODE REVIEW

File: scripts/utility/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This Readme explains the structure and process of passing the pool test cases from .sol to .cairo
```
Comment (rcatalan98) [inline scripts/utility/README.md:1]:
```suggestion
This Readme explains the structure and process of passing the pool test cases from the original Uniswap Protocol to a compatible and more readable cairo structure.
```
Also add a short explanation on why we are doing this.

File: crates/yas_core/src/tests/utils/pool_1.cairo [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            amount_1_delta: IntegerTrait::<i256>::new(1165774985123750584, false),
+            execution_price: 158590,
+            fee_growth_global_0_X128_delta: 0,
+            fee_growth_global_1_X128_delta: 595039006852697554786973994761078087,
+            pool_price_after: 250000,
+            pool_price_before: 100000,
+            tick_after: IntegerTrait::<i32>::new(9163, false),
+            tick_before: IntegerTrait::<i32>::new(0, false),
+        },
+        SwapExpectedResults { //
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/pool_1.cairo:183]:
```suggestion
        SwapExpectedResults {
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+            }
+            i += 1;
+        };
+        result
+    }
+
+    fn POOL_CASES() -> Array<PoolTestCase> {
+        array![
+            PoolTestCase {
+                // description: 'low fee, 1:1 price, 2e18 max range liquidity',
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/swap_cases.cairo:104]:
```suggestion
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo [hunk 4]
```diff
@@ -0,0 +1,505 @@
...
+                mint_positions: array![
+                    Position {
+                        tick_lower: get_min_tick(FeeAmount::LOW),
+                        tick_upper: get_max_tick(FeeAmount::LOW),
+                        liquidity: 2000000000000000000,
+                    },
+                ],
+            },
+            PoolTestCase {
+                // description: 'medium fee, 1:1 price, 2e18 max range liquidity',
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/swap_cases.cairo:116]:
```suggestion
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
+                mint_positions: array![
+                    Position {
+                        tick_lower: get_min_tick(FeeAmount::MEDIUM),
+                        tick_upper: get_max_tick(FeeAmount::MEDIUM),
+                        liquidity: 2000000000000000000,
+                    },
+                ],
+            },
+            PoolTestCase {
+                // description: 'high fee, 1:1 price, 2e18 max range liquidity',
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/swap_cases.cairo:128]:
```suggestion
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+                mint_positions: array![
+                    Position {
+                        tick_lower: get_min_tick(FeeAmount::HIGH),
+                        tick_upper: get_max_tick(FeeAmount::HIGH),
+                        liquidity: 2000000000000000000,
+                    },
+                ],
+            },
+            PoolTestCase {
+                // description: 'medium fee, 10:1 price, 2e18 max range liquidity',
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/swap_cases.cairo:140]:
```suggestion
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo [hunk 7]
```diff
@@ -0,0 +1,505 @@
...
+                mint_positions: array![
+                    Position {
+                        tick_lower: get_min_tick(FeeAmount::MEDIUM),
+                        tick_upper: get_max_tick(FeeAmount::MEDIUM),
+                        liquidity: 2000000000000000000,
+                    },
+                ],
+            },
+            PoolTestCase {
+                // description: 'medium fee, 1:10 price, 2e18 max range liquidity',
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/swap_cases.cairo:152]:
```suggestion
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo [hunk 8]
```diff
@@ -0,0 +1,505 @@
...
+                mint_positions: array![
+                    Position {
+                        tick_lower: get_min_tick(FeeAmount::MEDIUM),
+                        tick_upper: IntegerTrait::<i32>::new(tick_spacing(FeeAmount::MEDIUM), true),
+                        liquidity: 2000000000000000000,
+                    },
+                    Position {
+                        tick_lower: IntegerTrait::<i32>::new(
+                            tick_spacing(FeeAmount::MEDIUM), false
+                        ), //tick_spacing(FeeAmount::MEDIUM),
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/utils/swap_cases.cairo:176]:
```suggestion
                        ), 
```

File: crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo [hunk 9]
```diff
@@ -694,6 +717,210 @@ mod YASPoolTests {
...
+            expected_cases: Array<SwapExpectedResults>,
+            swap_cases: Array<SwapTestCase>
+        ) {
+            let mut i = 0;
+            assert(expected_cases.len() == swap_cases.len(), 'wrong amount of expected cases');
+            loop {
+                if i == expected_cases.len() {
+                    break;
+                }
+                'case'.print();
```
Comment (rcatalan98) [inline crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo:783]:
```suggestion
```

File: crates/yas_core/src/tests/utils/swap_cases.cairo (hunk 3 above)
Comment (uri-99) [reply crates/yas_core/src/tests/utils/swap_cases.cairo:104]: all these are the pool case description, are we sure we want to delete them?

File: crates/yas_core/src/tests/utils/pool_1.cairo [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+            amount_1_delta: IntegerTrait::<i256>::new(1000, true),
+            execution_price: 99502,
+            fee_growth_global_0_X128_delta: 680564733841876926926,
+            fee_growth_global_1_X128_delta: 0,
+            pool_price_after: 100000,
+            pool_price_before: 100000,
+            tick_after: IntegerTrait::<i32>::new(1, true),
+            tick_before: IntegerTrait::<i32>::new(0, false),
+        },
+        SwapExpectedResults {
```
Comment (dubzn) [inline crates/yas_core/src/tests/utils/pool_1.cairo:157]: ok

File: crates/yas_core/src/tests/utils/pool_1.cairo [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+            amount_1_delta: IntegerTrait::<i256>::new(1005, false),
+            execution_price: 100500,
+            fee_growth_global_0_X128_delta: 0,
+            fee_growth_global_1_X128_delta: 680564733841876926926,
+            pool_price_after: 100000,
+            pool_price_before: 100000,
+            tick_after: IntegerTrait::<i32>::new(0, false),
+            tick_before: IntegerTrait::<i32>::new(0, false),
+        },
+        SwapExpectedResults {
```
Comment (dubzn) [inline crates/yas_core/src/tests/utils/pool_1.cairo:170]: ok

File: crates/yas_core/src/tests/utils/pool_1.cairo [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+            amount_1_delta: IntegerTrait::<i256>::new(1165774985123750584, false),
+            execution_price: 158590,
+            fee_growth_global_0_X128_delta: 0,
+            fee_growth_global_1_X128_delta: 595039006852697554786973994761078087,
+            pool_price_after: 250000,
+            pool_price_before: 100000,
+            tick_after: IntegerTrait::<i32>::new(9163, false),
+            tick_before: IntegerTrait::<i32>::new(0, false),
+        },
+        SwapExpectedResults {
```
Comment (dubzn) [inline crates/yas_core/src/tests/utils/pool_1.cairo:183]: ok

**Review by dubzn — APPROVED** [review]
:fire: :fire:

File: crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo [hunk 13]
```diff
@@ -1518,6 +1532,209 @@ mod YASPoolTests {
...
         }
+
+        mod PoolCase1 {
+            use super::test_pool;
+            use yas_core::tests::utils::pool_1::{SWAP_CASES_POOL_1, SWAP_EXPECTED_RESULTS_POOL_1};
+            use yas_core::tests::utils::swap_cases::SwapTestHelper::{
+                PoolTestCase, SwapExpectedResults, obtain_swap_cases, POOL_CASES
+            };
+
+            use debug::PrintTrait;
```
Comment (dpinones) [inline crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo:1543]: remove import

**Pull request merged**